solutions/task_3.py

```diff
@@ -6,57 +6,63 @@
 from datetime import datetime, timedelta
 
 
-def add_2(func):
-    def wrapper(a, b):
-        return func(a, b) + 2
-    return wrapper
+CACHE = {}
+CACHE_LIFE_TIME = 1
 
 
-def add_n(n):
-    def wrapper_1(func):
-        def wrapper_2(a, b):
-            return func(a, b) + n
-        return wrapper_2
-    return wrapper_1
+def add_2(func):
+    def wrapped(*args, **kwargs):
+        return func(*args, **kwargs) + 2
+    return wrapped
 
 
-CACHE = {}
+def add_n(n):
+    def wrapper(func):
+        def wrapped(*args, **kwargs):
+            return func(*args, **kwargs) + n
+        return wrapped
+    return wrapper
 
 
 def cached(cache_life_time):
     def wrapper_1(func):
         CACHE[func] = {}
 
         def wrapper_2(*args, **kwargs):
-            if str(args) + str(kwargs) not in CACHE[func].keys():
-                CACHE[func][str(args) + str(kwargs)] = [None, None]
-                CACHE[func][str(args) + str(kwargs)][0] = func(*args, **kwargs)
-                CACHE[func][str(args) + str(kwargs)][1] = datetime.now()
-                return CACHE[func][str(args) + str(kwargs)][0]
-            if (datetime.now() - CACHE[func][str(args)+str(kwargs)][1]).total_seconds() < cache_life_time:
-                return CACHE[func][str(args)+str(kwargs)][0]
-            CACHE[func][str(args)+str(kwargs)][0] = func(*args, **kwargs)
-            CACHE[func][str(args)+str(kwargs)][1] = datetime.now()
-            return CACHE[func][str(args)+str(kwargs)][0]
+            key = str(args) + str(kwargs)
+            if key not in CACHE[func]:
+                CACHE[func][key] = [func(*args, **kwargs), time.time()]
+                return CACHE[func][key][0]
+            if (time.time() - CACHE[func][key][1]) < cache_life_time:
+                return CACHE[func][key][0]
+            else:
+                CACHE[func][key][0] = func(*args, **kwargs)
+                CACHE[func][key][1] = time.time()
+                return CACHE[func][key][0]
         return wrapper_2
     return wrapper_1
 
 
 class Contact(object):
     def __init__(self, phone, operator):
-        self._phone_ = phone
-        self.operator = operator
+        self._phone = phone
+        self._operator = operator
 
     @property
     def phone(self):
-        return "{} {} {}".format("".join(self._phone_[:4]), "".join(self._phone_[4:]), self.operator)
+        return "".join("".join(self._phone[:-7]) + " " + "".join(self._phone[-7:]) + " " + self.operator)
```

Comment:
Omg...
Please. Tell. Me. What. This. Code. Is. Doing? What. Is. Sense. Do. You. See. Here?
How are you able to write such code?
```
Hmmm... I have a string '1234567'
How can I convert it to the same string? Idea!
I can split it into chars -> ['1', '2', '3', '4', '5', '6', '7']
And join all of them with empty string!! "".join(['1', '2', '3', '4', '5', '6', '7'])
Great! I have got the same string '1234567'!
```